### PR TITLE
Editorial: Add `Iterator Record` type

### PR DESCRIPTION
**Pull request by ExE-Boss**
This adds a `<dfn>` and `<emu‑table>` for “[Iterator Records](https://ci.tc39.es/preview/tc39/ecma262/pull/2591#sec-iterator-records)”.

---

Part of <https://github.com/tc39/ecma262/issues/2486>

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -6978,6 +6978,60 @@ <h1>
     <h1>Operations on Iterator Objects</h1>
     <p>See Common Iteration Interfaces (<emu-xref href="#sec-iteration"></emu-xref>).</p>
 
+    <emu-clause id="sec-iterator-records">
+      <h1>Iterator Records</h1>
+      <p>A <dfn variants="Iterator Records">Iterator Record</dfn> is a Record value used to encapsulate an Iterator or AsyncIterator along with the `next` method.</p>
```
Comment (jmdyck) [inline spec.html:6983]:
```suggestion
      <p>An <dfn variants="Iterator Records">Iterator Record</dfn> is a Record value used to encapsulate an Iterator or AsyncIterator along with the `next` method.</p>
```

**Comment by michaelficarra** [comment]
Partially resolves #2486.

**Review by michaelficarra — APPROVED** [review]
LGTM

File: spec.html [hunk 2]
```diff
@@ -6978,12 +6978,66 @@ <h1>
...
+          </tr>
+          <tr>
+            <td>
+              [[Iterator]]
+            </td>
+            <td>
+              An object
+            </td>
+            <td>
+              An object that conforms to the <emu-xref href="#sec-iterator-interface">Iterator</emu-xref> or <emu-xref href="#sec-asynciterator-interface">AsyncIterator</emu-xref> interface.
```
Comment (bakkot) [inline spec.html:7006]:
```suggestion
              An object that conforms to the <em>Iterator</em> or <em>AsyncIterator</em> interface.
```

Right now we refer to these interface names with italics, and don't link them. https://github.com/tc39/ecma262/issues/2486 will fix that, but in the mean time we should be consistent with existing usage.

**Review by bakkot — APPROVED** [review]
LGTM other than comment.

**Review by jmdyck — DISMISSED** [review]
I just noticed that you could also change the 'type' of the `_iteratorRecord_` parameter from `unknown` to `an Iterator Record` in each of:
- IteratorBindingInitialization
- IteratorDestructuringAssignmentEvaluation
- ForIn/OfBodyEvaluation
- PerformPromiseAll
- PerformPromiseAllSettled
- PerformPromiseAny
- PerformPromiseRace

And similarly for `_syncIteratorRecord_` in `CreateAsyncFromSyncIterator`.

**Review by syg — APPROVED** [review]

**Pull request merged**